src/main/java/com/example/Game.java

```diff
@@ -58,19 +50,20 @@ public Game() {
                 System.out.println("Invalid symbol! Please choose X or O.");
             }
         }
-        player1 = new HumanPlayer(playerName, playerSymbol);
+        scanner.nextLine();
+
+        // Create human player using the factory method
+        player1 = PlayerFactory.createHumanPlayer(playerName, playerSymbol);
```

Comment:
Standard implementation of Factory DP is,
there should be only single method inside it, and you are suppose to pass type of object you need.

Check Below resource to know about standard implementation
- [Article](https://www.digitalocean.com/community/tutorials/factory-design-pattern-in-java)

Reply:
It is not mandatory have single method in factory class. I have used here multiple methods bcz as I wanted to apply SRP and also it makes the code more extensible. Now in future if I want to add the logic for creating CpuPlayer with extra configurations of AI then I don't have the need to change my whole PlayerFactory class and affect creation of HumanPlayer also.